src/Config.php

```diff
@@ -95,30 +97,18 @@ class Config implements ConfigInterface
      */
     public function __construct($options = [])
     {
-        if (! is_array($options) && ! ($options instanceof ArrayAccess)) {
-            throw new Exception\InvalidArgumentException(
-                'Config data must be of type array or ArrayAccess'
-            );
-        }
-
+        $this->ensureArrayOrArrayAccess($options);
         $this->preferences = $this->getDataFromArray($options, 'preferences')?: [];
         $this->types = $this->getDataFromArray($options, 'types')?: [];
     }
 
     /**
-     * @param array $data
-     * @param string $key
-     * @return array|ArrayAccess|null
+     * @param array|ArrayAccess $data
      */
-    private function getDataFromArray($data, $key)
+    private function getDataFromArray($data, string $key) : array
     {
-        if (! isset($data[$key])
-            || (! is_array($data[$key]) && ! ($data[$key] instanceof ArrayAccess))
-        ) {
-            return null;
-        }
-
-        return $data[$key];
+        $result = $data[$key] ?? [];
+        return is_array($result) ? $result : [];
```

Comment:
This function used to `return null` at some point, while it's no longer doing that. BC break or bug that wasn't spotted?

Reply:
It should not. The properties initialized by this method should never be null. If it does, the method at https://github.com/zendframework/zend-di/blob/master/src/Config.php#L208 fails

-> Potential bug spottet

Reply:
Good as-is IMO